scan/scan_context.go

```diff
@@ -10,7 +10,7 @@ type ScanContext struct {
 }
 
 func NewScanContext(scanID string) *ScanContext {
-	statusChan := make(chan bool)
+	statusChan := make(chan bool, 100)
```

Comment:
This will remove the deadlock but that solution seems wasteful?
I think we should keep the 0 bounded channel but keep the job alive until the job is truly terminated instead, that would be cleaner

Reply:
We don't have the deadlock in the SecretScanner (agent).
This buffer is added to avoid blocking the scanner routine while sending the data on channel.